src/mixins/DrawingTool.js

```diff
@@ -154,7 +155,16 @@ const DrawingTool = types
       startDrawing(x, y) {
         self.annotation.history.freeze();
         self.mode = "drawing";
-        const currentArea = self.createDrawingRegion(self.createRegionOptions({ x, y }));
+
+        const isPolygon = self.control.type.startsWith("polygon");
```

Comment:
the code here is good, my only small recommendation would be to create startDrawing in editor/src//tools/Polygon.js and extend the DrawingTool.startDrawing.
this way you dont need to do the check for isPolygon and just do a check for the FF

Reply:
The reason why I think that is the better way to go is because it properly encapsulates the behaviour difference in the correct location. Good suggestion 👍